public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -598,12 +614,55 @@ export class ElasticDatasource extends DataSourceApi<ElasticsearchQuery, Elastic
     );
   }
 
-  isMetadataField(fieldName: string) {
+  private legacy_useBrowserAccessMode() {
+    return !this.url.startsWith(`/api/datasources/proxy/${this.id}`);
```

Comment:
hmm seems weird, don't we just have a property for this in DS settings? Like the moment you use the backend implementation this won't hold because the url will change even though it will still be server access.

Reply:
>  don't we just have a property for this in DS settings?

I couldn't find a way to get this info, it seems like changing the access mode only changes the URL, there's no other `isBrowserAccess: boolean`-like property.

So, basically `this.url` is either the actual ES URL in browser mode or the proxy URL in Server mode. It will be ignored completely when completing the migration as it's hardcoded to be `/api/tsdb/query`. 
The idea is for "queries" that are ported to the new backend to fallback to the `legacy_*` methods only when in browser access mode.

that way we can migrate all the requests to the new BE implementation gradually, while supporting both Browser access mode and server access mode until all of them are migrated, then we can just remove all the `legacy_*` methods and remove the access mode setting.

Reply:
but anyway, this led me thinking that the URL might not start with `/api/datasources/proxy/` if we are serving grafana from a subpath. adding it to the todo list

Reply:
Hmm yeah got it seems lie we do not send it to FE with the `DataSourceInstanceSettings`. But maybe we could add it? I assume it wasn't needed before but I don't there is any downside in that? Like if we start migrate other DS we may need it even later to be able to do the migration iterativelly.